lib/matplotlib/rcsetup.py

```diff
@@ -60,8 +62,28 @@ def __call__(self, s):
                          % (self.key, s, list(six.itervalues(self.valid))))
 
 
+def _listify_validator(scalar_validator):
+    def f(s):
+        if isinstance(s, six.string_types):
+            try:
+                return [scalar_validator(v.strip()) for v in s.split(',')
+                        if v.strip()]
+            except Exception:
+                # Sometimes, a list of colors might be a single string
+                # of single-letter colornames. So give that a shot.
+                return [scalar_validator(v.strip()) for v in s if v.strip()]
+        elif type(s) in (list, tuple):
+            return [scalar_validator(v) for v in s if v]
+        else:
+            msg = "'s' must be of type [ string | list | tuple ]"
+            raise ValueError(msg)
+    f.__doc__ = scalar_validator.__doc__
+    return f
+
```

Comment:
Ahh yes, eek, not sure about the string, I think we should strip, because if someone writes "a,b,,c" then does that represent a typo or an empty string, that raises another question, how do we handle None?

I don't think we should allow for a list to get passed as rgbcmky, that feels wrong, we currently don't allow it, and it feels bad to put specific behaviour inside a generic function... maybe pass it in as extra argument, but still what about all the other colours which don't have a single letter attached to them?  See http://matplotlib.org/_images/named_colors.png for the complete list.

Yes I agree with you that if in doubt go private, but I do see it as a generic list validator that can use another validator to validate its items, but doesn't have to, we should allow it to just validate that we have a list...